config/generate_config.py

```diff
@@ -0,0 +1,792 @@
+parser.add_argument('--auto', action='store_true', help='Turn off interactive mode')
+# Define the '--no-remote-containers' flag
+parser.add_argument('--no-remote-containers', action='store_true', help='Disable container platform detection in remote partition')
+# Define the '--no-remote-devices' flag
+parser.add_argument('--no-remote-devices', action='store_true', help='Disable devices detection in remote partition')
+# Define the '--reservations' flag
+parser.add_argument('--reservations', nargs='?', 
+    help='Specify the reservations that you want to create partitions for.'
+)
+
+args = parser.parse_args()
+
+user_input = not args.auto
+
+containers_search = 'y'
+if args.no_remote_containers:
+    containers_search = 'n'
+
+devices_search = 'y'
+if args.no_remote_devices:
+    devices_search = 'n'
+
+if args.reservations:
+    reservations_based = args.reservations.split(',')
+else:
+    reservations_based = False
+
+user_input = not args.auto
+
+def main(user_input, containers_search, devices_search, reservations_based):
+
+    system_config['systems']   = []
+
+    # Get the system name 
+    cluster_name = os.getenv('CLUSTER_NAME')
+    if cluster_name:
+        logger.info(f'System name is {cluster_name}\n')
+    else:
+        cluster_name = 'cluster'
+        logger.warning(f'System name not found set to "{cluster_name}\n"')
+
+    system_config['systems'].append({'name': cluster_name})
+
+    # Get the hostname
+    try:
+        hostname = socket.gethostname() # Same method reframe uses
+    except Exception as e:
+        hostname = '<hostname>'
+        logger.error(f'Hostname not found')
+        logger.debug(f'Trying to retrieve the hostname raised:\n{e}\n')
+    else:
+        hostname = hostname.strip()
+        hostname = re.search(r'(^\w+)', hostname)
+        hostname = hostname.group(0)
+        logger.info(f'Hostname is {hostname}\n')
+
+    if hostname != '<hostname>' and cluster_name != 'cluster':  
+        if hostname != cluster_name:
+            logger.warning(f'Detected hostname and ' +
+                          f'systemname are different\n') 
+
+    system_config['systems'][0].update({'hostnames': [hostname]})
+
+    # Get the modules system
+    module_system = 'nomod'
+    # Detect lmod and tmod
+    module_info = os.getenv('LMOD_CMD')
+    if not module_info:
+        try:
+            module_info = subprocess.run(['modulecmd','-V'], 
+                stdout=subprocess.PIPE,stderr=subprocess.PIPE,
+                universal_newlines=True
+            )
+        except FileNotFoundError:
+            try:
+                module_info = subprocess.run(['spack','-V'], 
+                    stdout=subprocess.PIPE,stderr=subprocess.PIPE,
+                    universal_newlines=True
+                )
+            except FileNotFoundError:
+                pass
+            else:
+                module_system = 'spack'
+        else:
+            version_tmod = re.search(r'^VERSION=(\S+)', module_info.stderr,

```

Comment:
```suggestion
            version_tmod = re.search(r'^VERSION=(\S+)', module_info.stdout,

```
At least on the old daint the output is int stdout